spec/models/results/response_tree_builder_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "response tree" do
+  # form with repeat group
+  context "simple form" do
+    let(:form1) { create(:form, question_types: %w[text text text]) }
+
+    it "should produce a simple tree from a form with three children" do
+      response_tree = Results::ResponseTreeBuilder.new(form1).build
+      # same number of leaves at the root
+      expect(response_tree.root_answers.count).to eq form1.root_questionings.count
```

Comment:
Let's not add this root_answers method. It seems like kind of a smell to me. And let's not use root_questionings either. Instead, you can just do `response_tree.children` and `form_tree.children`. (Note that response_tree is really a root node just like root_group).

You then would add like a `let(:form_tree) { form1.root_group }`.